daffodil-lib/src/main/scala/org/apache/daffodil/schema/annotation/props/PropertyScoping.scala

```diff
@@ -131,6 +138,22 @@ trait FindPropertyMixin extends PropTypes {
           lr
         }
       }
+    /* if this looks like it could be an epression, issue a warning */
+    /* only if isForExpression is false*/
+    /* for runtime mixins, need to supply true, other places, no change*/
+    if( ! isForExpression ) {
+      propRes match{ 
+        case Found( v, _, _, _ ) => {
+
+          if( v.startsWith( "{")) {
```

Comment:
We should make this test a bit more selective, like check if it begins with double curly braces (which is an escape, so don't warn), and also check that it begins and ends with a single brace.

Reply:
To restate - we want to warn only if the value begins and ends with { }, and doesn't begin with {{. Is that your intent?

Reply:
Confirmed.

Reply:
DPathUtil has an ``isExpression`` function which I think has that exact logic. Probably makes sense to just reuse that so if what an expression looks like ever changes we only have to update that one function.

Reply:
Moved DPathUtil from daffodil-runtime1 to daffodil-lib and invoked isExpression rather than forming the check here.